### PR TITLE
Target .NET Standard 1.1

### PR DESCRIPTION
**Pull request by cbales**
This includes updating project files to xproj, adding .NET Core unit tests, and replacing DataMember attributes with JsonProperty notation. These changes make us .NET Standard 1.1 compliant, as requested by #49 
This is a minor release, bringing us to 1.3.0 for Microsoft.Graph and 1.4.0 for Microsoft.Graph.Core

### CODE REVIEW

**Comment by twsouthwick** [comment]
What's the purpose of the duplicated tests? You should be able to just retarget the project to build against whichever framework you need without duplicating the actual source files.

File: tests/Microsoft.Graph.DotnetCore.Core.Test/Helpers/ExtractSelectHelperTest.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        public void ArgumentNullOnNullArgument()
+        {
+            Assert.Throws<ArgumentNullException>(() => this.TestExtractMembers<Event>(null));
+        }
+
+        [Fact]
+        public void SingleMemberAnonymousTypeImplicitMember()
+        {
+            Expression<Func<Event, object>> expression = (theEvent) => new { theEvent.Body };
+            string s = this.TestExtractMembers(expression);
```
Comment (daboxu) [inline tests/Microsoft.Graph.DotnetCore.Core.Test/Helpers/ExtractSelectHelperTest.cs]: using var s = this.TestExtractMemebers()? I saw you use var in the StringHelperTest below, so better keep consistent style.

File: tests/Microsoft.Graph.DotnetCore.Core.Test/Helpers/StringHelperTests.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using Xunit;
+
+namespace Microsoft.Graph.DotnetCore.Core.Test.Helpers
+{
+    public class StringHelperTests
+    {
+        [Fact]
+        public void ConvertTypeToLowerCamelCase()
+        {
+            var expectedTypeString = "microsoft.graph.type";
```
Comment (daboxu) [inline tests/Microsoft.Graph.DotnetCore.Core.Test/Helpers/StringHelperTests.cs]: using const if it is the expected value

File: tests/Microsoft.Graph.DotnetCore.Core.Test/Requests/BaseClientTests.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        public BaseClientTests()
+        {
+            this.authenticationProvider = new MockAuthenticationProvider();
+        }
+
+        [Fact]
+        public void BaseClient_InitializeBaseUrlWithoutTrailingSlash()
+        {
+            var expectedBaseUrl = "https://localhost";
```
Comment (daboxu) [inline tests/Microsoft.Graph.DotnetCore.Core.Test/Requests/BaseClientTests.cs]: const

File: tests/Microsoft.Graph.DotnetCore.Core.Test/Requests/BaseRequestBuilderTests.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+using Xunit;
+
+namespace Microsoft.Graph.DotnetCore.Core.Test.Requests
+{
+    public class BaseRequestBuilderTests
+    {
+        [Fact]
+        public void BaseRequestBuilder()
+        {
+            var requestUrl = "https://localhost";
```
Comment (daboxu) [inline tests/Microsoft.Graph.DotnetCore.Core.Test/Requests/BaseRequestBuilderTests.cs]: const

File: tests/Microsoft.Graph.DotnetCore.Core.Test/Requests/RequestTestBase.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Graph.DotnetCore.Core.Test.Requests
+{
+    public class RequestTestBase : IDisposable
+    {
+        protected string baseUrl = "https://localhost/v1.0";
```
Comment (daboxu) [inline tests/Microsoft.Graph.DotnetCore.Core.Test/Requests/RequestTestBase.cs]: const

File: tests/Microsoft.Graph.DotnetCore.Test/Requests/Functional/ExcelTests.cs [hunk 6]
```diff
@@ -0,0 +1,532 @@
...
+                    Assert.True(false, "Something happened. Error code: " + e.Error.Code);
+                }
+            }
+
+            return "";
+        }
+
+        public async Task OneDriveUploadTestFileContent(string fileId)
+        {
+            //try
```
Comment (daboxu) [inline tests/Microsoft.Graph.DotnetCore.Test/Requests/Functional/ExcelTests.cs]: clean up commented tests?

File: tests/Microsoft.Graph.DotnetCore.Test/Requests/Functional/ExcelTests.cs [hunk 7]
```diff
@@ -0,0 +1,532 @@
...
+
+                await OneDriveDeleteTestFile(excelFileId, 3000);
+            }
+            catch (Microsoft.Graph.ServiceException e)
+            {
+                Assert.True(false, "The unprotect call failed to remove protection from the worksheet. Error code: " + e.Error.Code);
+            }
+        }
+
+        //public async Task ExcelTestTemplate()
```
Comment (daboxu) [inline tests/Microsoft.Graph.DotnetCore.Test/Requests/Functional/ExcelTests.cs]: clean up

File: tests/Microsoft.Graph.DotnetCore.Test/Requests/Functional/MailTests.cs [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+                Assert.NotNull(mailFolderMessagesCollectionPage);
+            }
+            catch (Microsoft.Graph.ServiceException e)
+            {
+                Assert.True(false, "Something happened, check out a trace. Error code: " + e.Error.Code);
+            }
+        }
+
+        //// Test that we can set an attachment on a mail, send it, and then retrieve it.
```
Comment (daboxu) [inline tests/Microsoft.Graph.DotnetCore.Test/Requests/Functional/MailTests.cs]: clean up

File: tests/Microsoft.Graph.DotnetCore.Test/Requests/Functional/OneDriveTests.cs [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Graph.DotnetCore.Test.Requests.Functional
+{
+    public class OneDriveTests : GraphTestBase
+    {
+        // https://github.com/OneDrive/onedrive-sdk-csharp/blob/master/docs/chunked-uploads.md
```
Comment (daboxu) [inline tests/Microsoft.Graph.DotnetCore.Test/Requests/Functional/OneDriveTests.cs]: clean up

**Comment by cbales** [comment]
@twsouthwick AFAIK only an alpha version of Moq supports .NET Core. We didn't want to change this dependency until it was supported on GA.

**Comment by twsouthwick** [comment]
That's fine - you can have different dependencies for different frameworks. See https://github.com/OfficeDev/Open-XML-SDK/blob/vNext/DocumentFormat.OpenXml/project.json for an example of this.

**Comment by cbales** [comment]
Thanks for the test code reviews, @daboxu . We are going to create a separate PR to merge the two test suites into one, at which point we'll apply your suggestions. This way, we don't have to make them in both test suites.

**Comment by daboxu** [comment]
👍

**Pull request merged**